win32 wheel build is broken

Probably caused by #316 cc @lpsinger 

Example log: https://github.com/astropy/astropy-healpix/actions/runs/28384215558/job/84094560015

```
Creating library build\...\astropy_healpix\_core.lib and object build\...\astropy_healpix\_core.exp
  _core.obj : error LNK2001: unresolved external symbol __BitScanReverse64
  build\lib.win32-cpython-310\astropy_healpix\_core.pyd : fatal error LNK1120: 1 unresolved externals
  error: command 'C:\\...\\x86\\link.exe' failed with exit code 1120
```

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -24,7 +24,6 @@ jobs:
       targets: |
         - cp3*-manylinux_x86_64
         - cp3*-musllinux_x86_64
-        - cp3*-win32
         - cp3*-win_amd64
         - cp3*-macosx_x86_64
         - cp3*-macosx_arm64

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -9,6 +9,8 @@ Changes
 
 - Require Numpy >= 2.
 
+- Wheel for ``win32`` is no longer provided. Use wheel for ``win_amd64`` instead.
+
 - Fix incorrect return values of ``uniq_to_level_ipix`` near npix boundaries.
 
 - The functions ``nside_to_level`` and ``uniq_to_level_ipix`` no longer raise

